test/release/examples/patterns/readcsv.chpl

```diff
@@ -0,0 +1,286 @@
+   The included metadata-20lines.csv only has 20 lines of data including the
+   header row that has the column names.
+
+   Each line in csv files is a row of data.  Columns are separated by commas.
+   Some of the cells of data have quotes because the data string itself includes
+   a comma or commas.
+
+   There are many options in terms of data structures that such data could be
+   read into.  Here are the two approaches illustrated here.
+       (1) pythonic -- list of Map instances where there is an entry in the Map
+               per column (key=column name, value=column value for that row)
+       (2) chapeltastic -- associative array with the key being the column name
+               and the value being an array of strings
+
+   You can compile and run this example by on the command line doing a
+       source util/quickstart/setchplenv.bash
+   in the chapel home directory, and then compiling and running this file as follows:
+       chpl --fast readcsv.chpl
+       ./readcsv
+
+  You can check that the output files for both approaches end up the same as the input
+  file by doing the following on the command line:
+      diff metadata-20lines.csv readcsv-out1.csv
+      diff metadata-20lines.csv readcsv-out2.csv
+  If you do not get any output then the files are the same and don't have any differences.
+
+  References:
+    file readers and writers
+      https://chapel-lang.org/docs/main/modules/standard/IO.html#i-o-overview
+      https://chapel-lang.org/docs/primers/fileIO.html
+
+    lists
+      https://chapel-lang.org/docs/primers/listOps.html
+
+    maps
+      https://chapel-lang.org/docs/main/modules/standard/Map.html
+ */
+use IO;
+use List;
+use Map;
+
+config const inFileName = "metadata-20lines.csv";
+config const approach1FileName = "readcsv-out1.csv";
+config const approach2FileName = "readcsv-out2.csv";
+config const debug = true;
+
+// Open up a file to work with.
+var f = open(inFileName, iomode.r);
+
+// Now read the in the csv data. 
+// Approach 1: formatted input into a list of maps
+{
+  writeln();
+  writeln();
+  writeln("Approach 1: formatted input into a list of maps");
+
+  var reader = f.reader();
+
+  // Read the first line to get the column names.
+  // Assuming the following format, where the number of columns (ncol) is not 
+  // known ahead of time:
+  //    col_name_1,col_name_2,...,col_name_ncol\n
+  var line : string;
+
+  if (!reader.readline(line)) then
+    writeln("ERROR: ", inFileName, " appears to be empty");
+
+  var colNames : list(string);
+  colNames = createListOfColNames(line);
+
+  if debug {
+    writeln("\ncolNames: ", colNames);
+  }
+
+  // Create a list of maps with one map per row of data.
+  var dataRows : list(map(string,string));
+  // Read individual lines from the file until the end of the file.
+  while (reader.readline(line)) {
+    // Create a new map per line.
+    var aRowMap = new map(string, string);
```

Comment:
Nit: there's some inconsistency about whether you put a space after a comma or not.

Reply:
fixed throughout file